server/controllers/refer/referralsController.ts

```diff
@@ -134,6 +134,27 @@ export default class ReferralsController {
     }
   }
 
+  reason(): TypedRequestHandler<Request, Response> {
+    return async (req: Request, res: Response) => {
+      TypeUtils.assertHasUser(req)
+
+      const referral = await this.referralService.getReferral(req.user.token, req.params.referralId)
+      const person = await this.personService.getPerson(req.user.token, referral.prisonNumber)
+
+      if (!person) {
+        throw createError(404, {
+          userMessage: `Person with prison number ${referral.prisonNumber} not found.`,
```

Comment:
Spotted some of the other methods incorrectly(?) refer to `req.params.prisonNumber` which I can update in another commit/PR.

Reply:
Good catch. I think these have been missed because we aren't testing the message - only the status code. We tried before, but the assertion would pass regardless of what string you pass in

Reply:
Good catch. I think these have been missed because we aren't testing the message - only the status code. We tried before, but the assertion would pass regardless of what string you pass in

Reply:
I think this might have something to do with specifying an object as the second parameter with `userMessage`.  What's the reason for doing it this way? Looking at [https-errors](https://github.com/jshttp/http-errors) readme, it looks like the error should just be passed as the second parameter.

Reply:
You're right, I think we've got slightly confused on how we should be handling errors in the application. I think we picked up this use of an object with a `userMessage` field in it from a couple of other projects, but they're using that to display the validation error to the user. We're not doing that anywhere.

I think we should refactor either:
- The code to use this userMessage variable
- The way we handle errors to pass a string (easier to test).

I'll add a card to tidy this up.

Reply:
For the sake of this PR I've added a test to check for the error with a string.  Can refactor the others in the future task